### ISSUE TITLE
Create a git wrapper that polyfill's unsupported behaviour

### ISSUE THREAD

**Issue by silvergasp** [Byron/gitoxide#1222]
### Summary 💡

Create a command line app that is completely git compatible. This command line app would determine if gitoxide can handle the command line request, and if not defer that to git. e.g.
```
# gitoxide can handle clones, so gitoxide does the clone.
gix-polyfill clone https://github.com/Byron/gitoxide.git

# gitoxide can't handle rebasing, so it forwards the command line args to git.
gix-polyfill rebase upstream/main
```

### Motivation 🔦

As much as I really enjoy blazingly fast clones and the `gix-*` API's, I don't end up using `gix` as much as I could. This is mostly down to a little bit of cognitive gymnastics e.g.
- Does gix support this command?
- Let's just try it and find out.
- Nope doesn't work.
- Let's just use git.

As this becomes muscle memory, I tend to reach towards git rather than gix. To the point where I use gix significantly less than I thought I would. This isn't really a problem with gitoxide itself as it's an enormous undertaking and a WIP. But having a hybrid `git`/`gix` would be a nice way to ease the transition while getting some of the nice performance benefits of gitoxide.

**Comment by Byron** [comment]
**tl;dr** I welcome the community to maintain such a tool with the option of eventual adoption into this codebase, should there be such desire.

----

I think the main misconception, probably also aided by the similarity of the word `gix` with `git`, is that `gix` isn't meant as daily driver. Its API is forever unstable (even though I'd think that major sub-commands like `clone` or one day `checkout` could become stable to allow their usage on CI), and it's meant to be nothing more than a developer tool to help trying `gix` functionality on real repositories.

When I see my own usage, I by now use it only to clone repos as it's genuinely faster and, by now, correct. But for everything else I use `git` and `stg` respectively. And I do so happily as these tools are great at their respective jobs and `gix` doesn't do anything better or more comfortably. Maybe one exception is `gix index entries` which I use more often to see which attributes are set, and `git` can't do it that well.

However, I do hope that in a couple of years when `gix` is stable and has (mostly) feature parity with `git2` I can finally deliver on a promise to myself and make `ein` the daily driver everyone would want to use.

Until then, `gix` is what it is and a 'gix powered git' is something the community would have to come up with and maintain. If that happens and it is indeed permanently maintained, I certainly wouldn't object to adopt it and its maintainers to this codebase.

On another note, Git, the project, is open to adopting Rust as well and who knows what that means for `gitoxide` - my hope would be that one day most of `git` will be powered by Rust, which then maybe can also be used as a library.
